includes/compat/subscriptions/class-wc-payment-gateway-wcpay-subscriptions-compat.php

```diff
@@ -0,0 +1,109 @@
+<?php
+/**
+ * Class WC_Payment_Gateway_WCPay_Subscriptions_Compat
+ *
+ * @package WooCommerce\Payments
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+use WCPay\Logger;
+
+/**
+ * Gateway class for WooCommerce Payments, with added compatibility with WooCommerce Subscriptions.
+ */
+class WC_Payment_Gateway_WCPay_Subscriptions_Compat extends WC_Payment_Gateway_WCPay {
+
+	/**
+	 * WC_Payment_Gateway_WCPay_Subscriptions_Compat constructor.
+	 *
+	 * @param array ...$args Arguments passed to the main gateway's constructor.
+	 */
+	public function __construct( ...$args ) {
+		parent::__construct( ...$args );
+
+		$this->supports = array_merge(
+			$this->supports,
+			[
+				'subscriptions',
+				'subscription_cancellation',
+				'subscription_suspension',
+				'subscription_reactivation',
+				'subscription_amount_changes',
+				'subscription_date_changes',
+				'subscription_payment_method_change',
+				'subscription_payment_method_change_customer',
+				'subscription_payment_method_change_admin',
+				'multiple_subscriptions',
+			]
+		);
+
+		add_action( 'woocommerce_scheduled_subscription_payment_' . $this->id, [ $this, 'scheduled_subscription_payment' ], 10, 2 );
+		add_filter( 'wcs_new_order_created', [ $this, 'copy_token_from_parent_order' ], 10, 2 );
+		add_action( 'woocommerce_subscription_failing_payment_method_updated_' . $this->id, [ $this, 'update_failing_payment_method' ], 10, 2 );
+		add_filter( 'wc_payments_display_save_payment_method_checkbox', [ $this, 'display_save_payment_method_checkbox' ], 10 );
+	}
+
+	public function process_payment( $order_id, $is_recurring_payment = false ) {
+		return parent::process_payment( $order_id, wcs_order_contains_subscription( $order_id ) );
+	}
+
+	public function display_save_payment_method_checkbox( $display ) {
+		if ( WC_Subscriptions_Cart::cart_contains_subscription() ) {
+			return false;
+		}
+		// Only render the "Save payment method" checkbox if there are no subscription products in the cart.
+		return $display;
+	}
+
+	/**
+	 * Scheduled_subscription_payment function.
+	 *
+	 * @param $amount float The amount to charge.
+	 * @param $renewal_order WC_Order A WC_Order object created to record the renewal payment.
+	 */
+	public function scheduled_subscription_payment( $amount, $renewal_order ) {
+		$order_tokens = WC_Payment_Tokens::get_order_tokens( $renewal_order );
```

Comment:
A token can be attached to an order, so I used that instead of polluting the order `meta` with WCPay-specific stuff.